src/lib/functions/runtimes/js/index.mjs

```diff
@@ -48,12 +51,52 @@ export const getBuildFunction = async ({ config, directory, errorExit, func, pro
   // main file otherwise.
   const functionDirectory = dirname(func.mainFile)
   const srcFiles = functionDirectory === directory ? [func.mainFile] : [functionDirectory]
-  const schedule = await parseForSchedule({ mainFile: func.mainFile, config, projectRoot })
 
-  return () => ({ schedule, srcFiles })
+  return () => ({ schedule: metadata.schedule, srcFiles })
 }
 
+const workerURL = new URL('worker.mjs', import.meta.url)
+
 export const invokeFunction = async ({ context, event, func, timeout }) => {
+  if (func.runtimeAPIVersion !== 2) {
+    return await invokeFunctionDirectly({ context, event, func, timeout })
+  }
+
+  const workerData = {
+    clientContext: JSON.stringify(context),
+    event,
+    // If a function builder has defined a `buildPath` property, we use it.
```

Comment:
nit: Since we're calling this `entryFilename` in zip-it-and-ship-it, might be nice to use the same naming here to make the relationship clear. Since this is the full path and not just the filename, perhaps `entryPath` or `entryFilePath` would be good?